cmd/manager/main.go

```diff
@@ -191,11 +186,8 @@ func main() {
 			),
 		).
 		WithAgentRegistrationOption(manager.NewRegistrationOption(nativeClient)).
-		WithAgentDeployTriggerClusterFilter(utils.ClusterImageRegistriesAnnotationChanged)
-
-	if agentInstallAll {
-		agentFactory.WithInstallStrategy(agent.InstallAllStrategy(common.AddonAgentInstallNamespace))
-	}
+		WithAgentDeployTriggerClusterFilter(utils.ClusterImageRegistriesAnnotationChanged).
+		WithAgentInstallNamespace(func(addon *v1alpha1.ManagedClusterAddOn) string { return common.AddonAgentInstallNamespace })
```

Comment:
`return common.AddonAgentInstallNamespace` will make the ns not configurable, I recommend we read the ns from AddonDeploymentConfig first, if it is not configured by an addondeploymentconfig, then we return the default `common.AddonAgentInstallNamespac`e value.

and there are some helper funcs you can use in the addon-framework repo. like https://github.com/open-cluster-management-io/addon-framework/blob/5c9527506035ef7a81c03af45e3dbc1597645fbc/cmd/example/helloworld_helm/main.go#L131C1-L134C6